### PR TITLE
implement listening for confirmation

### PR DESCRIPTION
**Pull request by Ioxom**
mostly working for now, but still needs stability and performance to be checked on

### CODE REVIEW

File: src/main/java/io/ioxcorp/ioxbox/Main.java [hunk 1]
```diff
@@ -41,14 +42,14 @@
         }
     }
```
Comment (Thonkman) [inline src/main/java/io/ioxcorp/ioxbox/Main.java:44]: fancy, makes it not changeable :wonk:
if you are willing, change the frame thing into changeable (I forget what its called)

File: src/main/java/io/ioxcorp/ioxbox/Main.java [hunk 2]
```diff
@@ -41,14 +42,14 @@
...
 
-    public static Frame frame;
+    public static final Frame frame;
     static {
         //create frame
         frame = new Frame();
         frame.init();
     }
 
-    public static HashMap<Long, Box> boxes;
```
Comment (Thonkman) [inline src/main/java/io/ioxcorp/ioxbox/Main.java:51]: makes it not changeable, very hot

**Review by Ioxom — COMMENTED** [review]
Should be good to merge now, just waiting on a review from @Thonkman .

File: src/main/java/io/ioxcorp/ioxbox/Main.java [hunk 3]
```diff
@@ -80,10 +81,10 @@ public static void main(String[] args) {
 
         //add event listeners
         if (api != null) {
-            api.addEventListener(new Listener());
```
Comment (Thonkman) [inline src/main/java/io/ioxcorp/ioxbox/Main.java:83]: finishes my unfinished work

File: src/main/java/io/ioxcorp/ioxbox/data/format/Box.java [hunk 4]
```diff
@@ -25,6 +25,7 @@
      * @param owner must be a {@link io.ioxcorp.ioxbox.data.format.CustomUser CustomUser} or {@link net.dv8tion.jda.api.entities.User User};
      * @param object must be a {@link java.lang.String String}, {@link io.ioxcorp.ioxbox.data.format.CustomUser CustomUser} or {@link net.dv8tion.jda.api.entities.User User}; the object that is inside the newly created box
      * @exception IllegalArgumentException if the owner or object parameters are incompatible types
+     * @author ioxom
```
Comment (Thonkman) [inline src/main/java/io/ioxcorp/ioxbox/data/format/Box.java:28]: add my name like rn

File: src/main/java/io/ioxcorp/ioxbox/data/format/Box.java [hunk 5]
```diff
@@ -62,6 +63,7 @@ public Box(Object owner, Object object) {
      * @param owner must be a {@link io.ioxcorp.ioxbox.data.format.CustomUser CustomUser} or {@link net.dv8tion.jda.api.entities.User User}; the owner of the box
      * @exception IllegalArgumentException if the owner object passed is an incompatible type
      * @see Box#Box(Object, Object)
+     * @author ioxom
```
Comment (Thonkman) [inline src/main/java/io/ioxcorp/ioxbox/data/format/Box.java:66]: same here

File: src/main/java/io/ioxcorp/ioxbox/data/format/Box.java [hunk 6]
```diff
@@ -80,6 +82,7 @@ public Box(Object owner) {
 
     /**
      * do not use this, it is a constructor used only for jackson to be able to parse Box objects from json
+     * @author ioxom
```
Comment (Thonkman) [inline src/main/java/io/ioxcorp/ioxbox/data/format/Box.java:85]: i will now use * to demonstrate this

File: src/main/java/io/ioxcorp/ioxbox/data/format/Box.java [hunk 7]
```diff
@@ -91,6 +94,7 @@ public String toString() {
     /**
      * a method for making a discord-sendable representation of a box
      * @return a {@link net.dv8tion.jda.api.entities.MessageEmbed MessageEmbed} that can be sent in discord showing the contents of the referenced box
+     * @author ioxom
```
Comment (Thonkman) [inline src/main/java/io/ioxcorp/ioxbox/data/format/Box.java:97]: *

File: src/main/java/io/ioxcorp/ioxbox/data/format/Box.java [hunk 8]
```diff
@@ -106,6 +110,7 @@ public MessageEmbed embed() {
 
     /**
      * @return a readable {@link java.lang.String String} of the items in the specified box
+     * @author ioxom
```
Comment (Thonkman) [inline src/main/java/io/ioxcorp/ioxbox/data/format/Box.java:113]: *

File: src/main/java/io/ioxcorp/ioxbox/data/format/Box.java [hunk 9]
```diff
@@ -126,6 +131,7 @@ public String itemsToString() {
 
     /**
      * @return a readable {@link java.lang.String String} of the users in the specified box
+     * @author ioxom
```
Comment (Thonkman) [inline src/main/java/io/ioxcorp/ioxbox/data/format/Box.java:134]: *

File: src/main/java/io/ioxcorp/ioxbox/data/format/Box.java [hunk 10]
```diff
@@ -148,7 +154,8 @@ public String usersToString() {
      * adds the Object passed to the {@link io.ioxcorp.ioxbox.data.format.Box} specified if it is a compatible format, then saves to json
      * @param object a {@link java.lang.String String}, a {@link io.ioxcorp.ioxbox.data.format.CustomUser CustomUser} or {@link net.dv8tion.jda.api.entities.User User} to be added to the referenced {@link io.ioxcorp.ioxbox.data.format.Box}
      * @exception IllegalArgumentException if the object passed in an incompatible type
-     * @see Box#remove(Object) 
+     * @see Box#remove(Object)
+     * @author ioxom
```
Comment (Thonkman) [inline src/main/java/io/ioxcorp/ioxbox/data/format/Box.java:158]: *

File: src/main/java/io/ioxcorp/ioxbox/data/format/Box.java [hunk 11]
```diff
@@ -169,6 +176,7 @@ public void add(Object object) {
      * @param object a {@link java.lang.String String}, a {@link io.ioxcorp.ioxbox.data.format.CustomUser CustomUser} or {@link net.dv8tion.jda.api.entities.User User} to be removed from the referenced {@link io.ioxcorp.ioxbox.data.format.Box}
      * @exception IllegalArgumentException if the object passed in an incompatible type
      * @see Box#add(Object)
+     * @author ioxom
```
Comment (Thonkman) [inline src/main/java/io/ioxcorp/ioxbox/data/format/Box.java:179]: *

File: src/main/java/io/ioxcorp/ioxbox/data/format/Box.java [hunk 12]
```diff
@@ -191,6 +199,7 @@ public void remove(Object object) {
      * @exception IllegalArgumentException if the owner or item objects are incompatible types
      * @exception InvalidParameterException if the owner already has a box
      * @see Box#createBox(Object)
+     * @author ioxom
```
Comment (Thonkman) [inline src/main/java/io/ioxcorp/ioxbox/data/format/Box.java:202]: *

File: src/main/java/io/ioxcorp/ioxbox/data/format/Box.java [hunk 13]
```diff
@@ -214,6 +223,7 @@ public static void createBox(Object owner, Object item) {
      * @exception IllegalArgumentException if the owner or item objects are incompatible types
      * @exception InvalidParameterException if the owner already has a box
      * @see Box#createBox(Object, Object)
+     * @author ioxom
```
Comment (Thonkman) [inline src/main/java/io/ioxcorp/ioxbox/data/format/Box.java:226]: *

File: src/main/java/io/ioxcorp/ioxbox/data/format/Box.java [hunk 14]
```diff
@@ -235,6 +245,7 @@ public static void createBox(Object owner) {
      * checks if a box contains the object specified
      * @param object must be a {@link io.ioxcorp.ioxbox.data.format.CustomUser CustomUser}, {@link net.dv8tion.jda.api.entities.User User} or {@link java.lang.String}; the object to be checked for
      * @return true, if the object is found; false, if the object is not found
+     * @author ioxom
```
Comment (Thonkman) [inline src/main/java/io/ioxcorp/ioxbox/data/format/Box.java:248]: *

Comment (Thonkman) [inline src/main/java/io/ioxcorp/ioxbox/data/format/Box.java:248]: *

File: src/main/java/io/ioxcorp/ioxbox/data/format/WhatAmIDoing.java [hunk 15]
```diff
@@ -0,0 +1,20 @@
+package io.ioxcorp.ioxbox.data.format;
```
Comment (Thonkman) [inline src/main/java/io/ioxcorp/ioxbox/data/format/WhatAmIDoing.java:1]: cool

File: src/main/java/io/ioxcorp/ioxbox/helpers/EmbedHelper.java [hunk 16]
```diff
@@ -20,11 +21,21 @@ public EmbedHelper(CustomUser user) {
         this.random = new Random();
     }
 
+    @JsonIgnore
```
Comment (Thonkman) [inline src/main/java/io/ioxcorp/ioxbox/helpers/EmbedHelper.java:24]: have some choccy milk

**Review by Thonkman — APPROVED** [review]
weqwewad

File: src/main/java/io/ioxcorp/ioxbox/data/format/Box.java (hunk 4 above)
Comment (Ioxom) [reply src/main/java/io/ioxcorp/ioxbox/data/format/Box.java:28]: did you write this?

File: src/main/java/io/ioxcorp/ioxbox/helpers/EmbedHelper.java (hunk 16 above)
Comment (Ioxom) [reply src/main/java/io/ioxcorp/ioxbox/helpers/EmbedHelper.java:24]: futureproofing
forgetting @JsonIgnore on a getter is a bad idea

**Pull request merged**